sxs/waveforms/alignment.py

```diff
@@ -549,3 +551,284 @@ def align4d(
     wa_prime = wa_prime.rotate(δSpin3)
 
     return optimum.cost, wa_prime, optimum
+
+def map_waveform_to_canonical_frame(wa, t_ref):
+    """Map waveform to canonical frame at t_ref; this amounts to
```

Comment:
Docstrings are supposed to have a single-line summary, followed by two newlines, then more description if needed.  This format will break some help tools.